hw_12/index.php

```diff
@@ -0,0 +1,241 @@
+<?php
+
+declare(strict_types=1);
+error_reporting(E_ALL);
+ini_set('display_errors', '1');
+
+/*///////////NORMAL LEVEL////////////*/
+
+/*
+//Task 1
+function message(): string
+{
+    return "«Hello, World!»";
+}
+
+print_r(message());
+*/
+
+
+//Task 2
+/*
+function leapYear(int $year): bool
+{
+    if ($year % 4 == 0) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+var_dump(leapYear(2022));
+*/
+
+
+//Task 3
+/*
+function abbrev(string $str): string
+{
+    $abbrevStr = implode('', array_diff_assoc(str_split(ucwords($str)), str_split(strtolower($str))));
+    return $abbrevStr;
+}
+
+print_r(abbrev('Cascading Style Sheets'));
+*/
+
+
+//Task 4
+/*
+function strReversed(string $strStart): string
+{
+    $strFinish = '';
+    // mb_strlen - получение длины строки; mb_substr - возвращает часть строки
+    for ($i = mb_strlen($strStart); $i >= 0; $i--) {
+        $strFinish .= mb_substr($strStart, $i, 1);// .= - объединение строк с присвоением результата
+    }
+    return $strFinish;
+}
+
+print_r(strReversed('Nastya Настя'));
+*/
+
+/*
+//Task 5
+function IsPrime($a): bool
+{
+    for ($i = 2; $i < $a; $i++) {
+        if ($a % $i == 0) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    if ($a == 0) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+var_dump(isPrime(6));
+*/
+
+/*
+//Task 6
+function wordLenFinish($str)
+{
+    $arrNewWord = explode(" ", $str,); //explode — Разбивает строку с помощью разделителя
+    $strFinish = rtrim(end($arrNewWord), '!@#$%^&*()_+"№;%:?=-');
+//rtrim-убирает пробелы и заданные символы справа  // end-выводит последний элемент массива
+    echo iconv_strlen($strFinish); //iconv_strlen — Возвращает количество символов в строке
+}
+
+wordLenFinish('задание №6 PHP!');
+*/
+
+/*
+//Task 7
+function maxNumArr($arr)
+{
+    if (1 === count($arr)) {
+        return $arr;
+    }
+    $result = [];
+    foreach ($arr as $key => $item) {
+        foreach (maxNumArr(array_diff_key($arr, array($key => $item))) as $newArr)
+            //array_diff_key — Вычисляет расхождение массивов, сравнивая ключи
+        {
+            $result[] = $item . $newArr;
+        }
+    }
+    return $result;
+}
+
+$arr = [3, 24, 4];
+echo max(maxNumArr($arr));
+*/
+
+/*
+//Task 8
+function anagramma($word1, $word2): bool
+{
+    $arr1 = str_split($word1);
+    $arr2 = str_split($word2);
+    $w1 = strlen($word1);
+    $w2 = strlen($word2);
+
+    return ((!array_diff($arr1, $arr2)) and ($w1 == $w2));
```

Comment:
and не используем, используем &&
```suggestion
    return !array_diff($arr1, $arr2) && ($w1 == $w2);
```